### DISCUSSION TITLE
Apply sql hint nolock or ability to rename table name in sqlquery (repository)

### DISCUSSION THREAD

**Discussion by minhhungit** [serenity-is/Serenity#6333] [Help]
Hi team, does anyone know how to apply sql **nolock** hint into row.cs or repository

Another issue, I have a task which the grid is using 2 tables depend on quickfilter, for example either "dbo.TableA" and "dbo.TableBWithMoreDetails"
when I tried to call From() in Repository.PrepareQuery it said something like duplicate "From" so I have to generate 2 rows.cs vs 2 repositories and switch endpoint call. It works but have "duplicate" code.

So, is there a way to change name of table in From() ?

Any hint will help a lot, thank you

**Comment by dfaruque** [comment]
You could override `CreateQuery`

```
            protected override SqlQuery CreateQuery()
            {
                return new SqlQuery().Dialect(Connection.GetDialect()).From(Row);
            }

```

**Reply by minhhungit** [reply]
unlucky it does not work, how to change `Row` with row of another table
I tried to create the query like this but it throws some errors, finnally it called correct sql, grid show rows but EMPTY data.

I have to create new row.cs and repository to handle.

```csharp
protected override SqlQuery CreateQuery()
{
    //return new SqlQuery().Dialect(Connection.GetDialect()).From(Row);

    var t0 = Entities.AnotherTableRow.Fields.As("t0");
    return new SqlQuery().Dialect(Connection.GetDialect()).From(t0);
}

protected override void SelectField(SqlQuery query, Field field)
{
    if (field.Name != fld.ZipCode.Name)
    {
        base.SelectField(query, field);
    }
}
```

Btw, do you have any suggest about how to apply **sql hint**, thank you

**Reply by dfaruque** [reply]
Sorry. I don't know about  **sql hint**

**Reply by minhhungit** [reply]
maybe to apply sql hint we have to wrap the table by using sql view
